mediatek: add Nokia EA0326GMP support

1715173329 opened the pull request:
```
Hardware specification:
  SoC: MediaTek MT7981B 2x A53
  Flash: 128 MB SPI-NAND
  RAM: 256MB
  Ethernet: 4x 10/100/1000 Mbps
  Switch: MediaTek MT7531AE
  WiFi: MediaTek MT7976C
  Button: Reset, WPS/Mesh
  Power: DC 12V 1A

Gain SSH access:
1. Login into web interface, and download the configuration.
2. Download the configration utilities:
   https://firmware.download.immortalwrt.eu.org/cnsztl/mediatek/filogic/openwrt-mediatek-mt7981-nokia-ea0326gmp-config-utils.tar.gz
     These binaries are extraced from the factory firmware, which are
     dynamically linked with aarch64 musl 1.1.24. To use them, you
     must run them under the same runtime environment, otherwise the
     binaries will not work properly!
3. Upload the configuration and utilities to a suitable environment.
4. Uncompress the utilities, move them to '/bin' and give them executable permisison:
   tar -zxf openwrt-mediatek-mt7981-nokia-ea0326gmp-config-utils.tar.gz
   mv mkconfig seama /bin
   chmod +x /bin/mkconfig
   chmod +x /bin/seama
5. Decrypt and uncompress the configuration:
     Enter fakeroot if you are not login as root.
   mkconfig -a de-enca -m EA0326GMP_3FE79221BAAA -i EA0326GMP_3FE79221BAAA-xxxxxxxx-backup.tar.gz -o backup.tar.gz
   tar -zxf backup.tar.gz
6. Edit 'etc/config/dropbear', set 'enable' to '1'.
7. Edit 'etc/passwd', remove root password: 'root::1:0:99999:7:::'.
8. Repack the configuration:
   tar -zcf backup.tar.gz etc/
   mkconfig -a enca -m EA0326GMP_3FE79221BAAA -i backup.tar.gz -o EA0326GMP_3FE79221BAAA-xxxxxxxx-backup.tar.gz
9. Upload new configuration via web interface, now you can SSH to EA0326GMP.

A minimum configuration which enabled SSH access is also provided
to simplify the process:
https://firmware.download.immortalwrt.eu.org/cnsztl/mediatek/filogic/openwrt-mediatek-mt7981-nokia-ea0326gmp-enable-ssh.tar.gz

Flash instructions:
1. SSH to EA0326GMP, backup everything, especially 'Factory' part.
2. Write new BL2:
   mtd write openwrt-mediatek-filogic-nokia_ea0326gmp-preloader.bin BL2
3. Write new FIP:
   mtd write openwrt-mediatek-filogic-nokia_ea0326gmp-bl31-uboot.fip FIP
4. Set static IP on your PC:
   IP 192.168.1.254/24, GW 192.168.1.1
5. Serve OpenWrt initramfs image using TFTP server.
6. Cut off the power and re-engage, wait for TFTP recovery to complete.
7. After OpenWrt has booted, run the following command:
     This is required as there is a bug about detecting rootdisk.
   sed -i 's,ubiblock,,g' /lib/upgrade/platform.sh
8. Perform sysupgrade.
```

1715173329 committed:
mediatek: add Nokia EA0326GMP support

Hardware specification:
  SoC: MediaTek MT7981B 2x A53
  Flash: 128 MB SPI-NAND
  RAM: 256MB
  Ethernet: 4x 10/100/1000 Mbps
  Switch: MediaTek MT7531AE
  WiFi: MediaTek MT7976C
  Button: Reset, WPS/Mesh
  Power: DC 12V 1A

Gain SSH access:
1. Login into web interface, and download the configuration.
2. Download the configration utilities:
   https://firmware.download.immortalwrt.eu.org/cnsztl/mediatek/filogic/openwrt-mediatek-mt7981-nokia-ea0326gmp-config-utils.tar.gz
     These binaries are extraced from the factory firmware, which are
     dynamically linked with aarch64 musl 1.1.24. To use them, you
     must run them under the same runtime environment, otherwise the
     binaries will not work properly!
3. Upload the configuration and utilities to a suitable environment.
4. Uncompress the utilities, move them to '/bin' and give them executable permisison:
   tar -zxf openwrt-mediatek-mt7981-nokia-ea0326gmp-config-utils.tar.gz
   mv mkconfig seama /bin
   chmod +x /bin/mkconfig
   chmod +x /bin/seama
5. Decrypt and uncompress the configuration:
     Enter fakeroot if you are not login as root.
   mkconfig -a de-enca -m EA0326GMP_3FE79221BAAA -i EA0326GMP_3FE79221BAAA-xxxxxxxx-backup.tar.gz -o backup.tar.gz
   tar -zxf backup.tar.gz
6. Edit 'etc/config/dropbear', set 'enable' to '1'.
7. Edit 'etc/passwd', remove root password: 'root::1:0:99999:7:::'.
8. Repack the configuration:
   tar -zcf backup.tar.gz etc/
   mkconfig -a enca -m EA0326GMP_3FE79221BAAA -i backup.tar.gz -o EA0326GMP_3FE79221BAAA-xxxxxxxx-backup.tar.gz
9. Upload new configuration via web interface, now you can SSH to EA0326GMP.

A minimum configuration which enabled SSH access is also provided
to simplify the process:
https://firmware.download.immortalwrt.eu.org/cnsztl/mediatek/filogic/openwrt-mediatek-mt7981-nokia-ea0326gmp-enable-ssh.tar.gz

Flash instructions:
1. SSH to EA0326GMP, backup everything, especially 'Factory' part.
2. Write new BL2:
   mtd write openwrt-mediatek-filogic-nokia_ea0326gmp-preloader.bin BL2
3. Write new FIP:
   mtd write openwrt-mediatek-filogic-nokia_ea0326gmp-bl31-uboot.fip FIP
4. Set static IP on your PC:
   IP 192.168.1.254/24, GW 192.168.1.1
5. Serve OpenWrt initramfs image using TFTP server.
6. Cut off the power and re-engage, wait for TFTP recovery to complete.
7. After OpenWrt has booted, perform sysupgrade.

Signed-off-by: Tianling Shen <cnsztl@immortalwrt.org>

1715173329 commented:
Hi @dangowrt could you please take a look?

For the first-time installation the `ubiblock_xx` device does not exist in `/sys/class/block`, so `fitblk_get_bootdev` outputs nothing, resulting sysupgrade failure.

musashino205 commented on package/boot/uboot-mediatek/Makefile:
```diff
@@ -267,6 +267,18 @@ define U-Boot/mt7981_jcg_q30-pro
 endef
 
+define U-Boot/mt7981_nokia_ea0326gmp
+  NAME:=Qihoo 360T7
```
wrong name

1715173329 replied:
fixed, thanks!

dangowrt commented:
> For the first-time installation the `ubiblock_xx` device does not exist in `/sys/class/block`, so `fitblk_get_bootdev` outputs nothing, resulting sysupgrade failure.

This is difficult to fix by design, and I guess the best would be to include `ubimkvol` and `ubiblock` calls in the installation instructions, so then the block device will exist and `fitblk_get_bootdev` should work. I'm open for other ideas, if any, of course.

1715173329 commented:
Would it make sense to simply introduce fitblk to legacy fit sysupgrade logic? The code will check if fit device exists so it should not break anything. And it's also unnecessary to do bootdev detection for a single-storage device.
```patch
	h3c,magic-nx30-pro|\
	jcg,q30-pro|\
	mediatek,mt7981-rfb|\
	netcore,n60|\
	qihoo,360t7|\
	xiaomi,mi-router-ax3000t-ubootmod|\
	xiaomi,mi-router-wr30u-ubootmod)
+		[ -e /dev/fit0 ] && fitblk /dev/fit0
+		[ -e /dev/fitrw ] && fitblk /dev/fitrw
		CI_KERNPART="fit"
		nand_do_upgrade "$1"
		;;
```

Tian-Jionglu commented:
> 7. After OpenWrt has booted, run the following command:
>      This is required as there is a bug about detecting rootdisk.
>    sed -i 's,ubiblock,,g' /lib/upgrade/platform.sh

This solve the problem posted in immortalwrt repo, thanks.

dangowrt commented:
> Would it make sense to simply introduce fitblk to legacy fit sysupgrade logic?

Technically this can work. I was kinda intending to make it as obvious as possible which device has already been converted to fitblk, so device owners notice and may carry out conversion on the device.

> And it's also unnecessary to do bootdev detection for a single-storage device.

The hope was kinda to have a single generic sysupgrade codepath, but yes, given the chicken-egg/bootstrapping problem when installing OpenWrt to a device currently running vendor firmware or older OpenWrt that would be the best work-around I can imagine.

Tian-Jionglu commented:
@1715173329 with latest build [ImmortalWrt SNAPSHOT r29875-80c7834062](https://firmware-selector.immortalwrt.org/?version=SNAPSHOT&target=mediatek%2Ffilogic&id=nokia_ea0326gmp), radio1 doesn't work on 160MHz, only works on 80MHz.
Could you please take a look?

1715173329 commented:
@Tian-Jionglu can you stop ending off-topic comments here?

1715173329 committed:
mediatek: filogic: move nand-only devices out of bootdev detection

For the first-time installation (mostly migrates from vendor firmware)
the ubiblock is not ready, but bootdev detection relies on it. This
means users must create ubiblock manually otherwise the sysupgrade
will not work.

Now a unique case is added for nand devices which use new fit format,
let's move to it.

Signed-off-by: Tianling Shen <cnsztl@immortalwrt.org>

1715173329 commented:
> I was kinda intending to make it as obvious as possible which device has already been converted to fitblk, so device owners notice and may carry out conversion on the device.

I see it, thank you!
I created a new 'case' for these devices as a workaround for now.

1715173329 commented:
@dangowrt could you please take a look? thanks!

1715173329 commented:
That's weird. I extracted this value from vendor dts. Will confirm it later.

1715173329 commented:
Indeed. So they just copy&paste the rfb dts. Thanks for report.
Would you mind leaving a `Reported-by` tag?

1715173329 commented:
#15651

iiipod commented:
> ```
> Hardware specification:
>   SoC: MediaTek MT7981B 2x A53
>   Flash: 128 MB SPI-NAND
>   RAM: 256MB
>   Ethernet: 4x 10/100/1000 Mbps
>   Switch: MediaTek MT7531AE
>   WiFi: MediaTek MT7976C
>   Button: Reset, WPS/Mesh
>   Power: DC 12V 1A
> 
> Gain SSH access:
> 1. Login into web interface, and download the configuration.
> 2. Download the configration utilities:
>    https://firmware.download.immortalwrt.eu.org/cnsztl/mediatek/filogic/openwrt-mediatek-mt7981-nokia-ea0326gmp-config-utils.tar.gz
>      These binaries are extraced from the factory firmware, which are
>      dynamically linked with aarch64 musl 1.1.24. To use them, you
>      must run them under the same runtime environment, otherwise the
>      binaries will not work properly!
> 3. Upload the configuration and utilities to a suitable environment.
> 4. Uncompress the utilities, move them to '/bin' and give them executable permisison:
>    tar -zxf openwrt-mediatek-mt7981-nokia-ea0326gmp-config-utils.tar.gz
>    mv mkconfig seama /bin
>    chmod +x /bin/mkconfig
>    chmod +x /bin/seama
> 5. Decrypt and uncompress the configuration:
>      Enter fakeroot if you are not login as root.
>    mkconfig -a de-enca -m EA0326GMP_3FE79221BAAA -i EA0326GMP_3FE79221BAAA-xxxxxxxx-backup.tar.gz -o backup.tar.gz
>    tar -zxf backup.tar.gz
> 6. Edit 'etc/config/dropbear', set 'enable' to '1'.
> 7. Edit 'etc/passwd', remove root password: 'root::1:0:99999:7:::'.
> 8. Repack the configuration:
>    tar -zcf backup.tar.gz etc/
>    mkconfig -a enca -m EA0326GMP_3FE79221BAAA -i backup.tar.gz -o EA0326GMP_3FE79221BAAA-xxxxxxxx-backup.tar.gz
> 9. Upload new configuration via web interface, now you can SSH to EA0326GMP.
> 
> A minimum configuration which enabled SSH access is also provided
> to simplify the process:
> https://firmware.download.immortalwrt.eu.org/cnsztl/mediatek/filogic/openwrt-mediatek-mt7981-nokia-ea0326gmp-enable-ssh.tar.gz
> 
> Flash instructions:
> 1. SSH to EA0326GMP, backup everything, especially 'Factory' part.
> 2. Write new BL2:
>    mtd write openwrt-mediatek-filogic-nokia_ea0326gmp-preloader.bin BL2
> 3. Write new FIP:
>    mtd write openwrt-mediatek-filogic-nokia_ea0326gmp-bl31-uboot.fip FIP
> 4. Set static IP on your PC:
>    IP 192.168.1.254/24, GW 192.168.1.1
> 5. Serve OpenWrt initramfs image using TFTP server.
> 6. Cut off the power and re-engage, wait for TFTP recovery to complete.
> 7. After OpenWrt has booted, run the following command:
>      This is required as there is a bug about detecting rootdisk.
>    sed -i 's,ubiblock,,g' /lib/upgrade/platform.sh
> 8. Perform sysupgrade.
> ```

请教一下：刷了官方openwrt的snapshot版本后有三个wifi是怎么回事，其中一个2.4g,另外两个是5G的，其中一个5G的手机搜索不到。请问官方固件本来就是这样的吗，还是哪里出了问题？
